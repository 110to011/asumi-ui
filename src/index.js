/**
 * Created by Elly on 2016/5/27.
 */

import Menu from './component/menu';
import SubMenu from './component/menu/subMenu';
import MenuItem from './component/menu/menuItem';
import MenuItemGroup from './component/menu/menuItemGroup';
import Tag from './component/tag';
import Grid from './component/grid';
import Form from './component/form';
import Tabs from './component/tabs';
import Radio from './component/radio';
import Group from './component/group';
import Input from './component/input';
import Modal from './component/modal';
import Table from './component/table';
import Editor from './component/editor';
import Col from './component/table/col';
import Button from './component/button';
import Select from './component/select';
import Loading from './component/loading';
import Popover from './component/popover';
import Message from './component/message';
import Animate from './component/animate';
import Upload from './component/upload';
import Checkbox from './component/checkbox';
import Dropdown from './component/dropdown';
import Option from './component/select/option';
import Pagination from './component/pagination';
import FormItem from './component/form/formItem';
import TabPanel from './component/tabs/tabPanel';
import RadioGroup from './component/radio/radioGroup';
import CheckboxGroup from './component/checkbox/checkGroup';
import SimplePagination from './component/pagination/simplePagination';
import DateTime from './component/datetime';

const Tooltip = Popover;

Table.Col = Col;

Select.Option = Option;

Tabs.TabPanel = TabPanel;

Menu.SubMenu = SubMenu;
Menu.MenuItem = MenuItem;
Menu.MenuItemGroup = MenuItemGroup;

Radio.Group = RadioGroup;
Checkbox.Group = CheckboxGroup;

Form.FormItem = FormItem;

Pagination.Simple = SimplePagination;

export {
    Col,
    Tag,
    Tabs,
    Menu,
    Form,
    Grid,
    Table,
    Modal,
    Input,
    Group,
    Radio,
    Select,
    Button,
    Upload,
    Option,
    Editor,
    SubMenu,
    Animate,
    Loading,
    Popover,
    Message,
    Tooltip,
    FormItem,
    MenuItem,
    TabPanel,
    Dropdown,
    Checkbox,
    DateTime,
    Pagination,
    RadioGroup,
    CheckboxGroup,
    MenuItemGroup,
    SimplePagination,
};