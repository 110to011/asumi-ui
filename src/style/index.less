@charset "utf-8";

@import "variables";
@import "normalize";
@import "fontawesome/less/font-awesome";

body {
  font-family: '微软雅黑', "Microsoft YaHei", Arial, Microsoft YaHei, Helvetica, sans-serif;
  margin: 0;
  color: @normalText;
  font-size: @fontSize;
  line-height: 1.42857;
}

a {
  color: @primary;
  &:hover, &:active, &:visited, &:focus {
    color: @d_primary;
  }
}

.el-shadow {
  box-shadow: 0 6px 12px @shadow;
  -webkit-box-shadow: 0 6px 12px @shadow;
}

.overflow {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@import "nav";
@import "tabs";
@import "drop";
@import "caret";
@import "modal";
@import "input";
@import "table";
@import "button";
@import "stretch";
@import "popover";
@import "checkbox";
@import "pagination";
