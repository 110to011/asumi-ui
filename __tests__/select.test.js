import React, {Component} from 'react';
import {shallow, mount} from 'enzyme';
import {Select, Option} from '../src';

describe('Select', () => {
    it('create component', () => {
        const component = mount(<Select/>);
        expect(component).toMatchSnapshot();
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define disabled', () => {
        const component = mount(<Select disabled/>);
        expect(component).toMatchSnapshot();
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define size[small]', () => {
        const component = mount(<Select size="small"/>);
        expect(component).toMatchSnapshot();
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define size[large]', () => {
        const component = mount(<Select size="large"/>);
        expect(component).toMatchSnapshot();
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define className', () => {
        const component = shallow(<Select className="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('.jest').length).toEqual(1);
    });
    it('define style', () => {
        const component = shallow(<Select style={{width: 100}}/>);
        expect(component).toMatchSnapshot();
    });
    it('define closeAfterSelect', () => {
        const component = shallow(<Select closeAfterSelect={false}/>);
        expect(component).toMatchSnapshot();
    });
    it('define name', () => {
        const component = mount(<Select name="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('input[name="jest"]').length).toEqual(1);
    });
    it('define readOnly', () => {
        const component = mount(<Select readOnly/>);
        expect(component).toBeDefined(); //喵喵？
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define searchable', () => {
        const component = mount(<Select searchable/>);
        expect(component).toBeDefined();
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define options', () => {
        const component = mount(
            <Select>
                <Option label="1" value={1}/>
                <Option label="2" value={2}/>
                <Option value={3}>3</Option>
                <Option label="4" value={4} disabled/>
            </Select>);
        const data = [{
            value: 1,
            disabled: undefined,
            label: "1"
        }, {
            value: 2,
            disabled: undefined,
            label: "2"
        }, {
            value: 3,
            disabled: undefined,
            label: "3"
        }, {
            value: 4,
            disabled: true,
            label: "4"
        }];
        expect(component.instance().state.data).toEqual(data);
    });
});

describe('Select change', () => {
    describe('change', () => {
        beforeAll(() => {
            jest.useFakeTimers();
        });

        let container;
        beforeEach(() => {
            container = document.createElement('div');
            document.body.appendChild(container);
        });

        afterAll(() => {
            jest.useRealTimers();
        });

        afterEach(() => {
            document.body.removeChild(container);
        });

        it('options rendered', () => {
            const component = mount(<Select/>, {attachTo: container});
            expect(component.instance().container).toBeUndefined();
            component.instance().handleToggle(true);
            expect(component.instance().state.visible).toBe(true);
            expect(component.instance().container).toBeDefined();
            expect(component.instance().el_select_menu).toBeDefined();
            expect(component.instance().el_select_ul).toBeDefined();
        });
        it('dropdownStyle', () => {
            const component = mount(<Select dropdownStyle={{width: 100}}/>, {attachTo: container});
            component.instance().handleToggle(true);
            expect(component.instance().el_select_menu.style.width).toBe("100px");
        });
        it('dropdownClassName', () => {
            const component = mount(<Select dropdownClassName="jest"/>, {attachTo: container});
            component.instance().handleToggle(true);
            expect(component.instance().el_select_menu.classList.contains("jest")).toBe(true);
        });
        it('closeAfterSelect[true]', () => {
            const component = mount(<Select closeAfterSelect><Option
                value={1}>1</Option></Select>, {attachTo: container});
            component.instance().handleToggle(true);
            component.instance().el_select_ul.children[0].click();
            expect(component.instance().state.visible).toBe(false);
        });
        it('closeAfterSelect[false]', () => {
            const component = mount(<Select closeAfterSelect={false}><Option
                value={1}>1</Option></Select>, {attachTo: container});
            component.instance().handleToggle(true);
            component.instance().el_select_ul.children[0].click();
            expect(component.instance().state.visible).toBe(true);
        });
        it('onChange', () => {
            class Demo extends Component {
                constructor() {
                    super();
                    this.state = {value: ""}
                }

                handleChange({value}) {
                    this.setState({value});
                }

                render() {
                    return (
                        <Select ref={c => this._c = c}
                                value={this.state.value}
                                onChange={this.handleChange.bind(this)}
                        >
                            <Option value={1}>1</Option>
                        </Select>
                    )
                }
            }

            const component = mount(<Demo/>, {attachTo: container});
            component.find('input').simulate('click');
            component.instance()._c.el_select_ul.children[0].click();
            expect(component.instance().state.value).toBe(1);
        });

        class Demo extends Component {
            constructor() {
                super();
                this.state = {value: []}
            }

            handleChange({value}) {
                this.setState({value});
            }

            render() {
                return (
                    <Select
                        multiple
                        {...this.props}
                        ref={c => this._c = c}
                        value={this.state.value}
                        onChange={this.handleChange.bind(this)}
                    >
                        <Option value={1}>1</Option>
                        <Option value={2}>2</Option>
                    </Select>
                )
            }
        }

        it('multiple', () => {
            const wrapper = mount(<Demo/>, {attachTo: container});
            wrapper.find('input').simulate('click');
            wrapper.instance()._c.el_select_ul.children[0].click();
            expect(wrapper.instance().state.value).toEqual([1]);
            wrapper.find('input').simulate('click');
            wrapper.instance()._c.el_select_ul.children[1].click();
            expect(wrapper.instance().state.value).toEqual([1, 2]);
            wrapper.find('input').simulate('click');
            wrapper.instance()._c.el_select_ul.children[1].click();
            expect(wrapper.instance().state.value).toEqual([1]);
        });
        it('selectAll', () => {
            const wrapper = mount(<Demo selectAll/>);
            wrapper.find('input').simulate('click');
            wrapper.instance()._c.el_select_ul.children[0].click();
            jest.runAllTimers();
            expect(wrapper.instance().state.value).toEqual([1, 2]);
            wrapper.find('input').simulate('click');
            wrapper.instance()._c.el_select_ul.children[0].click();
            expect(wrapper.instance().state.value).toEqual([]);
        });
        it('onSearch & onMatch', () => {
            const wrapper = mount(<Demo/>, {attachTo: container});
            const handleMatch = jest.fn();
            const handleSearch = jest.fn();
            wrapper.setProps({selectAll: true, searchable: true, onSearch: handleSearch, onMatch: handleMatch});
            wrapper.find('input').simulate('change', {target: {value: "a"}});
            jest.runAllTimers();
            expect(handleSearch).toBeCalled();
            expect(handleMatch).toBeCalled();
            jest.runAllTimers();
            expect(wrapper.instance()._c.el_select_ul.children[0].classList.contains('el-select-no-data')).toEqual(true);
        });
    });
});
