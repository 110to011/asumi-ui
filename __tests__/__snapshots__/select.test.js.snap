// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select create component 1`] = `
<Select
  closeAfterSelect={true}
  defaultValue=""
  noMatchText="暂无匹配数据"
  onChange={[Function]}
  onSearch={[Function]}
  selectAllText="全选"
  value=""
>
  <div
    className="el-select-wrapper"
  >
    <Input
      autoComplete="off"
      icon={
        <i
          className="el-caret"
        />
      }
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onPressEnter={[Function]}
      readOnly={true}
      type="text"
      value=""
    >
      <div
        className="el-input"
      >
        <span
          className="el-input-icon"
          onClick={[Function]}
        >
          <i
            className="el-caret"
          />
        </span>
        <input
          autoComplete="off"
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </Input>
  </div>
</Select>
`;

exports[`Select define closeAfterSelect 1`] = `
<div
  className="el-select-wrapper"
>
  <Input
    autoComplete="off"
    icon={
      <i
        className="el-caret"
      />
    }
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onPressEnter={[Function]}
    readOnly={true}
    type="text"
    value=""
  />
</div>
`;

exports[`Select define disabled 1`] = `
<Select
  closeAfterSelect={true}
  defaultValue=""
  disabled={true}
  noMatchText="暂无匹配数据"
  onChange={[Function]}
  onSearch={[Function]}
  selectAllText="全选"
  value=""
>
  <div
    className="el-select-wrapper"
  >
    <Input
      autoComplete="off"
      disabled={true}
      icon={
        <i
          className="el-caret"
        />
      }
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onPressEnter={[Function]}
      readOnly={true}
      type="text"
      value=""
    >
      <div
        className="el-input"
      >
        <span
          className="el-input-icon"
          onClick={[Function]}
        >
          <i
            className="el-caret"
          />
        </span>
        <input
          autoComplete="off"
          disabled={true}
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </Input>
  </div>
</Select>
`;

exports[`Select define size[large] 1`] = `
<Select
  closeAfterSelect={true}
  defaultValue=""
  noMatchText="暂无匹配数据"
  onChange={[Function]}
  onSearch={[Function]}
  selectAllText="全选"
  size="large"
  value=""
>
  <div
    className="el-select-wrapper el-large"
  >
    <Input
      autoComplete="off"
      icon={
        <i
          className="el-caret"
        />
      }
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onPressEnter={[Function]}
      readOnly={true}
      size="large"
      type="text"
      value=""
    >
      <div
        className="el-input el-large"
      >
        <span
          className="el-input-icon"
          onClick={[Function]}
        >
          <i
            className="el-caret"
          />
        </span>
        <input
          autoComplete="off"
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </Input>
  </div>
</Select>
`;

exports[`Select define size[small] 1`] = `
<Select
  closeAfterSelect={true}
  defaultValue=""
  noMatchText="暂无匹配数据"
  onChange={[Function]}
  onSearch={[Function]}
  selectAllText="全选"
  size="small"
  value=""
>
  <div
    className="el-select-wrapper el-small"
  >
    <Input
      autoComplete="off"
      icon={
        <i
          className="el-caret"
        />
      }
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onPressEnter={[Function]}
      readOnly={true}
      size="small"
      type="text"
      value=""
    >
      <div
        className="el-input el-small"
      >
        <span
          className="el-input-icon"
          onClick={[Function]}
        >
          <i
            className="el-caret"
          />
        </span>
        <input
          autoComplete="off"
          onChange={[Function]}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          readOnly={true}
          type="text"
          value=""
        />
      </div>
    </Input>
  </div>
</Select>
`;

exports[`Select define style 1`] = `
<div
  className="el-select-wrapper"
  style={
    Object {
      "width": 100,
    }
  }
>
  <Input
    autoComplete="off"
    icon={
      <i
        className="el-caret"
      />
    }
    onChange={[Function]}
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyPress={[Function]}
    onPressEnter={[Function]}
    readOnly={true}
    type="text"
    value=""
  />
</div>
`;
